### PR TITLE
[libjpeg-turbo] use d suffix for debug libs

### PR DESCRIPTION
**Pull request by cenit**
Using the d suffix in debug, it allows the official (cmake 3.13) FindJPEG to correctly find the two different configuration libraries

### CODE REVIEW

**Comment by Rastaban** [comment]
I am seeing failures in qt5-base on windows, it may need updated to find the new debug library name.

File: ports/libjpeg-turbo/portfile.cmake [hunk 1]
```diff
@@ -51,11 +51,14 @@ vcpkg_configure_cmake(
...
 # Rename libraries for static builds
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "static" AND EXISTS "${CURRENT_PACKAGES_DIR}/lib/jpeg-static.lib")
+if(EXISTS "${CURRENT_PACKAGES_DIR}/lib/jpeg-static.lib")
     file(RENAME "${CURRENT_PACKAGES_DIR}/lib/jpeg-static.lib" "${CURRENT_PACKAGES_DIR}/lib/jpeg.lib")
     file(RENAME "${CURRENT_PACKAGES_DIR}/lib/turbojpeg-static.lib" "${CURRENT_PACKAGES_DIR}/lib/turbojpeg.lib")
-    file(RENAME "${CURRENT_PACKAGES_DIR}/debug/lib/jpeg-static.lib" "${CURRENT_PACKAGES_DIR}/debug/lib/jpeg.lib")
-    file(RENAME "${CURRENT_PACKAGES_DIR}/debug/lib/turbojpeg-static.lib" "${CURRENT_PACKAGES_DIR}/debug/lib/turbojpeg.lib")
+    file(RENAME "${CURRENT_PACKAGES_DIR}/debug/lib/jpeg-static.lib" "${CURRENT_PACKAGES_DIR}/debug/lib/jpegd.lib")
+    file(RENAME "${CURRENT_PACKAGES_DIR}/debug/lib/turbojpeg-static.lib" "${CURRENT_PACKAGES_DIR}/debug/lib/turbojpegd.lib")
+else()
```
Comment (Rastaban) [inline ports/libjpeg-turbo/portfile.cmake]: There are build failures on osx/Linux that would probably be fixed if this `else()` were changed to elseif(EXISTS "${CURRENT_PACKAGES_DIR}/debug/lib/jpeg.lib")

**Comment by cenit** [comment]
very glad that you can find regressions and failures in other ports.
Shall I check qt5-base also?

File: ports/libjpeg-turbo/portfile.cmake (hunk 1 above)
Comment (cenit) [reply ports/libjpeg-turbo/portfile.cmake]: ok fair point, modifying with your suggestions right now

**Comment by Rastaban** [comment]
@cenit it would be great if you would be able to look into the qt5-base build failure.

**Comment by cenit** [comment]
ok sure, I'll try my best. Starting a build right now

**Comment by cenit** [comment]
ok I should have fixed the problem, please let me know if you find any other problem

**Comment by NancyLi1013** [comment]
@cenit 
Can we handle this via vcpkg-cmake-wrapper to solve this issue?

Since we recommend to use the name that upstream supports instead of renaming it.

Also this will block pkgconfig fixup. So I'm not sure which solution is more suitable.

What do you think?

**Comment by cenit** [comment]
That's the name that's most convenient for debug, and in fact the official cmake module looks for the d suffix... Maybe we can use a patch in the source code instead of renaming it after build?

**Pull request merged**